### ISSUE TITLE
AspNetIdentity Sample Says Created Users have wrong Username/Password

### ISSUE THREAD

**Issue by farzad99** [IdentityServer/IdentityServer3.Samples#216]
I ran the AspNetIdentity sample and created a few users using the admin feature of the sample. Then I ran the Clients sample with the "Javascript Implicit Client - Simple" sample set as the starting project. I clicked on the Get Token button which brought up the Login page. I tried the logins for all the created users but they all failed. I looked at the code and put a breakpoint on the following method:

public override async Task AuthenticateLocalAsync(LocalAuthenticationContext ctx)

but the breakpoint is never reached and the user is not authenticated with the error of user name/password is not correct. 

Could somebody tell me please why this is not working? I have looked at the IIS Express Icon and those two web sites are the only ones that are running and also I see the users exist in the AspNetUsers table assciated with the AspNetIdentity sample.

**Comment by brockallen** [comment]
If you have multiple copies of VS open all running IIS Express on the same port, then the last one you open wins. All of our samples that host IdSvr use the same port. Perhaps you're getting routed to the wrong one? Are you running multiple copies of VS with different samples hosting IdSvr?

**Comment by farzad99** [comment]
Thank you for your reply, Brock. The cause of it is not what you mentioned. So in my case, the login page is displayed by the IdentityServer and I enter valid user name and password. But I get "Error: Invalid username or password". But the code never reaches the "AuthenticateLocalAsync" method that's in the  "IdentityServer3.AspNetIdentity.cs" file. Are there any circumstances that a login is stated as invalid by the IdentityServer3 without even calling the "AuthenticateLocalAsync" method. I have a breakpoint on that function that's never hit.

**Comment by brockallen** [comment]
Not sure then. Check the IdSvr logs. If that turns up nothing, then the next thing is to debug into the source.

**Comment by farzad99** [comment]
OK, thanks. I'll do that and if I find out anything I'll post it here.

**Comment by farzad99** [comment]
I figured out what was wrong. When you use the same port in IIS Express for different projects then IIS Express gets confused. All the IdSvr3 hosts use port 44333. So I had used a few of IdSvr3 server projects with the same port. Even though, I had only one instance of the IdSver3 running in IIS Express the code was not for the correct project. Just change the port for different IdSvr3 and that fixes the issue.
